source/extensions/config_subscription/grpc/grpc_stream.h

```diff
@@ -80,7 +80,19 @@ class GrpcStream : public GrpcStreamInterface<RequestProto, ResponseProto>,
 
   bool grpcStreamAvailable() const override { return stream_ != nullptr; }
 
-  void sendMessage(const RequestProto& request) override { stream_->sendMessage(request, false); }
+  void sendMessage(const RequestProto& request) override {
+    control_plane_stats_.discovery_requests_.inc();
+    stream_->sendMessage(request, false);
+  }
+
+  void maybeRecordRequestStats(uint64_t payload_size, uint64_t resource_count) override {
```

Comment:
Wanted to try doing this accounting in `sendMessage` however since `RequestProto` is generic, things like `resource_count` can't be extracted out.